### PR TITLE
Feat/error handling api calls

### PR DESCRIPTION
**Pull request by raz-joan**
What (if any) features are you implementing?
- Error handling in API calls
- Add error displays in user views for failed fetches
- Add new fetch for specific game to improve performance on PairedPage
- Updates to cypress testing on Home page
- 
What (if anything) did you refactor?
- fetch calls
Were there any issues that arose?
- some issues with the data-cy in cypress tests after implementing an intercept
- 
Is there anything that you need from your teammate?
- update the tests for the homepage?
- 
Any other comments, questions, or concerns?
- Yehaw!

### CODE REVIEW

File: src/Components/Loading.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import '../Styles/Loading.css'
 const Loading = () => {
     return (
         <div className='loading-container'>
-            <h2>LOADING ...</h2>
+            <h2>LOADING . . .</h2>
```
Comment (colganmeanor) [inline src/Components/Loading.js:7]: this is the most important change in this branch.

File: src/App.css [hunk 2]
```diff
@@ -2,4 +2,16 @@
 .landing-page {
   background-color: #19435A;
   height: 532px;
+}
+
+.error-message-home {
+  position: fixed;
```
Comment (michelecomfort) [inline src/App.css:8]: I really like the Loading... error message in large font on middle of screen! Nicely done.

File: src/apiCalls.js [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
+                if (res.ok) {
+                    return res.json()
+                } else {
+                    throw new Error()
+                }
+            })
+            .catch(err => err)
+    },
+
+    getSpecificGame: (gameId) => {
```
Comment (colganmeanor) [inline src/apiCalls.js:15]: So helpful to have these all in this file and be able to call methods on the apiCalls object. Thank you Joan!

File: src/App.js [hunk 4]
```diff
@@ -13,23 +13,36 @@ import Favorites from './Components/Favorites'
...
+      })
   }, [])
 
   return (
     <div>
         <Header />
-        <main className='landing-page'>
+      <main className='landing-page'>
+        { error ?
+          <p className='error-message-home'>Sorry, there's been an error: '{error}'</p> :
```
Comment (michelecomfort) [inline src/App.js:39]: Good use interpolating in the error message

**Review by colganmeanor — COMMENTED** [review]
Fantastic work Joan - thank you so much for all the hard work! 🙏🏻

File: src/Components/PairedPage.js [hunk 5]
```diff
@@ -9,56 +9,55 @@ import apiCalls from '../apiCalls'
...
+                    <h2 className='perfect-pairing-title'>Your Perfect Pairing!</h2>
+                    <div className='paired-components'>
+                        {game && <Game />}
+                        <p className='plus-sign'>+</p>
+                        {currentDrink && <Drink />}
+                    </div>
+                    <div className='bottom-buttons'>
+                        <NavLink to={'/'} style={{ textDecoration: 'none' }}>
+                            <button className='return-button'>Return to Main</button>
+                        </NavLink>
```
Comment (michelecomfort) [inline src/Components/PairedPage.js:54]: I'll make sure you change these NavLinks to Links. Thanks for the call out!

File: cypress/integration/homepage.js [hunk 6]
```diff
@@ -47,32 +127,64 @@ describe('Board Game Bartender Home Page', () => {
...
             .should('have.value', 'Scotch')
 
-        cy.get('[data-cy=liquor-choice] option').eq(7)
+        cy.get('#liquors option').eq(7)
             .should('have.value', 'Non Alcoholic')
     })
-})
+
+    it('should reflect the correct value when typing in the game dropdown', () => {
+        cy.get('[data-cy=game-choice]')
```
Comment (michelecomfort) [inline cypress/integration/homepage.js:171]: Nice testing! Sweet suites friend!

**Review by michelecomfort — COMMENTED** [review]
Almost there!! ( clap! clap! clap! )

**Pull request merged**